src/liballoc/arc.rs

```diff
@@ -10,35 +10,11 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
-//! Threadsafe reference-counted boxes (the `Arc<T>` type).
+//! Thread-safe reference-counting pointers.
 //!
-//! The `Arc<T>` type provides shared ownership of an immutable value through
-//! atomic reference counting.
+//! See the [`Arc<T>`][arc] documentation for more details.
 //!
-//! `Weak<T>` is a weak reference to the `Arc<T>` box, and it is created by
```

Comment:
So, it's true that duplication isn't great. Usually, module-level docs should have an overview of the whole module, but with modules like these, it's just a container for Arc/Weak, and nothing else. As such, I think I'm fine with removing this in this instance. Especially given the `alloc/std` split as you mentioned.